src/Discord.Net.WebSocket/DiscordSocketClient.cs

```diff
@@ -1270,13 +1270,17 @@ private async Task ProcessMessageAsync(GatewayOpCode opCode, int? seq, string ty
                                             return;
                                         }
 
+                                        var cacheablelist = new List<Cacheable<IMessage, ulong>>();
```

Comment:
This would probably be better as an immutable list IMO.

Reply:
In addition, the lib's consensus is to have it be `ImmutableArray<T>`, specifically.